### PR TITLE
New keyboard: Rotr

### PR DESCRIPTION
**Pull request by kwlodarczy**
Adding a firmware entry for the PCB included with the ROTR group buy [geekhack](https://geekhack.org/index.php?topic=107104.0)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rotr/Makefile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ifndef MAKEFILE_INCLUDED
```
Comment (zvecr) [inline keyboards/rotr/Makefile:1]: This file should be deleted.

File: keyboards/rotr/config.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
+#ifndef CONFIG_H
+#define CONFIG_H
+
```
Comment (zvecr) [inline keyboards/rotr/config.h:3]:
```suggestion
#pragma once

```

File: keyboards/rotr/config.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* number of backlight levels */
+
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCING_DELAY 5
```
Comment (zvecr) [inline keyboards/rotr/config.h:41]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/rotr/config.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 0
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+
+#define TAP_CODE_DELAY 10
+#endif
+
+#endif
```
Comment (zvecr) [inline keyboards/rotr/config.h:68]:
```suggestion
```

File: keyboards/rotr/kb.c [hunk 5]
```diff
@@ -0,0 +1 @@
+#include "kb.h"
```
Comment (zvecr) [inline keyboards/rotr/kb.c:1]: This file should be renamed `rotr.c`, also...
```suggestion
#include "rotr.h"
```

File: keyboards/rotr/kb.h [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+#include "quantum.h"
+
+#define KEYMAP( \
+	K00, K01, K02  \
+) { \
+	{ K00,   K01,   K02 }  \
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/rotr/kb.h:12]: This file should be renamed `rotr.h`, also...

```suggestion
#pragma once

#include "quantum.h"

#define LAYOUT( \
	K00, K01, K02  \
) { \
	{ K00,   K01,   K02 }  \
}

```

File: keyboards/rotr/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		
+	}
+
+	if (usb_led & (1 << USB_LED_KANA)) {
+		
+	} else {
+		
+	}
+}
+
```
Comment (zvecr) [inline keyboards/rotr/keymaps/default/keymap.c:97]:
```suggestion
```

File: keyboards/rotr/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+	KEYMAP(
+		KC_TRNS, KC_TRNS, KC_TRNS),
+
+	KEYMAP(
+		KC_TRNS, KC_TRNS, KC_TRNS),
+
+	KEYMAP(
+		KC_TRNS, KC_TRNS, KC_TRNS)
+
```
Comment (zvecr) [inline keyboards/rotr/keymaps/default/keymap.c:52]:
```suggestion
	LAYOUT(
		KC_MPRV, KC_MPLY, KC_MNXT)
```

File: keyboards/rotr/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,109 @@
+#include "kb.h"
```
Comment (zvecr) [inline keyboards/rotr/keymaps/default/keymap.c:1]:
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/rotr/rules.mk [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+MOUSEKEY_ENABLE ?= yes	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
+RGBLIGHT_ENABLE ?= no
+ENCODER_ENABLE = yes	#Enables Rotary Encoder support
```
Comment (zvecr) [inline keyboards/rotr/rules.mk:57]: Ive had to guess the bootloader here

```suggestion
# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
ENCODER_ENABLE = yes        # Enables Rotary Encoder support
```

File: keyboards/rotr/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+	if (clockwise) {
+
+		tap_code(KC_VOLU);
+
+	} else {
+
+		tap_code(KC_VOLD);
+
+	}
+}
```
Comment (fauxpark) [inline keyboards/rotr/keymaps/default/keymap.c:20]:
```suggestion
    LAYOUT(
        KC_MPRV, KC_MPLY, KC_MNXT
    )
};

void encoder_update_user(uint8_t index, bool clockwise) {
    if (clockwise) {
        tap_code(KC_VOLU);
    } else {
        tap_code(KC_VOLD);
    }
}
```

File: keyboards/rotr/README.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
+# ROTR
```
Comment (fauxpark) [inline keyboards/rotr/README.md:1]: Please rename this file to lowercase.

File: keyboards/rotr/readme.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+### Features:
+* Minimal design with plug and play functionality.
+* Frictionless, solid brass knob.
+* Hot swappable switches.
+* USB C connection.
+
+Make example for this keyboard (after setting up your build environment):
+
+    make rotr:default:flash
```
Comment (fauxpark) [inline keyboards/rotr/readme.md:17]:
```suggestion
    make rotr:default
```

File: keyboards/rotr/readme.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
+# ROTR
+![ROTR](https://i.imgur.com/051liRC.jpg)
```
Comment (fauxpark) [inline keyboards/rotr/readme.md:2]:
```suggestion

![ROTR](https://i.imgur.com/051liRC.jpg)
```

File: keyboards/rotr/readme.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+
+A macro pad with a multipurpose knob.
+
+* Keyboard Maintainer: [kwlodarczyk](https://github.com/kwlodarczy)
+* Hardware: [Geekhack](https://geekhack.org/index.php?topic=107104.0)
+
+### Features:
```
Comment (fauxpark) [inline keyboards/rotr/readme.md:9]:
```suggestion
## Features:
```

File: keyboards/rotr/info.json [hunk 16]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "rotr", 
+    "url": "https://github.com/kwlodarczy/qmk_firmware/tree/master/keyboards/rotr",
```
Comment (Erovia) [inline keyboards/rotr/info.json:3]: [`A URL to the keyboard’s product page, QMK.fm/keyboards page, or other page describing information about the keyboard.`](https://docs.qmk.fm/#/reference_info_json)

File: keyboards/rotr/readme.md [hunk 17]
```diff
@@ -0,0 +1,20 @@
+# ROTR
+
+![ROTR](https://i.imgur.com/051liRC.jpg)
```
Comment (noroadsleft) [inline keyboards/rotr/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![ROTR](https://i.imgur.com/051liRCl.jpg)
```

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Pull request merged**